### PR TITLE
Hashmap, fixes

### PR DESCRIPTION
**Pull request by Riley31415**
Changed calculator to use hashmap, added tab completions, added long parse checks, fixed Issue#8

### CODE REVIEW

File: src/main/java/io/github/seggan/sfcalc/Calculator.java [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 import me.mrCookieSlime.CSCoreLibPlugin.cscorelib2.inventory.ItemUtils;
 import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.SlimefunItem;
+import me.mrCookieSlime.Slimefun.cscorelib2.inventory.ItemUtils;
```
Comment (Seggan) [inline src/main/java/io/github/seggan/sfcalc/Calculator.java]: Duplicate import, use cscorelib

File: src/main/java/io/github/seggan/sfcalc/Calculator.java [hunk 2]
```diff
@@ -30,76 +31,50 @@
...
+            }
         }
 
         return result;
     }
 
-    static void printResults(List<String> results, CommandSender sender, String s, SlimefunItem item, Long amount, SFCalc plugin) {
-        Set<String> resultSet = new HashSet<>(results);
-
+    static void printResults(HashMap<String, Long> results, CommandSender sender, String s, SlimefunItem item, Long amount, SFCalc plugin) {
```
Comment (Seggan) [inline src/main/java/io/github/seggan/sfcalc/Calculator.java]: Missing `HashMap` import, in other places as well

**Review by Seggan — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/seggan/sfcalc/CalcExecutor.java [hunk 3]
```diff
@@ -85,14 +84,22 @@ public boolean onCommand(CommandSender sender, Command command, String s, String
...
+                try {
+                    amount = Long.parseLong(args[1]);
+                    if (amount == 0) {
+                        sender.sendMessage(plugin.invalidNumberString);
+                        return true;
+                    }
+                } catch (NumberFormatException e) {
+                    sender.sendMessage(plugin.invalidNumberString);
+                    return true;
+                }
```
Comment (Seggan) [inline src/main/java/io/github/seggan/sfcalc/CalcExecutor.java]: Use the regex pattern check instead of try-catch

**Review by Seggan — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/seggan/sfcalc/CalcExecutor.java (hunk 3 above)
Comment (Riley31415) [reply src/main/java/io/github/seggan/sfcalc/CalcExecutor.java]: I cant figure out this matcher thing at all :(

Comment (Seggan) [reply src/main/java/io/github/seggan/sfcalc/CalcExecutor.java]: look at existing code

**Comment by Riley31415** [comment]
image

File: src/main/java/io/github/seggan/sfcalc/CalcExecutor.java (hunk 3 above)
Comment (Riley31415) [reply src/main/java/io/github/seggan/sfcalc/CalcExecutor.java]: The numeric matcher will not work for this, I need to check if the value is greater than long max value not whether it's a number

**Comment by Seggan** [comment]
Whats ur point with that image

File: src/main/java/io/github/seggan/sfcalc/CalcExecutor.java (hunk 3 above)
Comment (Seggan) [reply src/main/java/io/github/seggan/sfcalc/CalcExecutor.java]: ok then

**Comment by Riley31415** [comment]
show that it works i guess, and the formatting with the number in front of the name of item being calc

File: src/main/java/io/github/seggan/sfcalc/Calculator.java [hunk 4]
```diff
@@ -43,11 +43,19 @@ public Calculator(SFCalc plugin) {
...
         Map<String, Long> results = calculate(item);
 
-        sender.sendMessage(String.format(plugin.headerString, ChatColor.stripColor(ItemUtils.getItemName(item.getItem()))));
+        String header;
+        String name = ChatColor.stripColor(ItemUtils.getItemName(item.getItem()));
+        if (amount == 1) {
+            header = String.format(plugin.headerString, name);
+        } else {
+            header = String.format(plugin.headerAmountString, amount, name);
+        }
```
Comment (Seggan) [inline src/main/java/io/github/seggan/sfcalc/Calculator.java]: I said on discord

File: src/main/resources/config.yml [hunk 5]
```diff
@@ -21,4 +24,7 @@ category-error-string: "&cThat many categories is not supported yet. Please use
 item-error-string: "&cThat many items is not supported yet. Please use the command form of the calculator."
 # Not a player string ("You must be a player to send this message!").
 # Defaults to "&cYou must be a player to send this message!"
-not-a-player-string: "&cYou must be a player to send this message!"
\ No newline at end of file
+not-a-player-string: "&cYou must be a player to send this message!"
+# Invalid number string ("Invalid number!").
+# Defaults to "&cInvalid number!"
+invalid-number-string: "&cInvalid number!"
```
Comment (Seggan) [inline src/main/resources/config.yml]: Newline pls

**Review by Seggan — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/seggan/sfcalc/Calculator.java (hunk 4 above)
Comment (Seggan) [reply src/main/java/io/github/seggan/sfcalc/Calculator.java]: use my built in format here

**Review by Seggan — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/seggan/sfcalc/Calculator.java (hunk 4 above)
Comment (Riley31415) [reply src/main/java/io/github/seggan/sfcalc/Calculator.java]: but it workssss

Comment (Seggan) [reply src/main/java/io/github/seggan/sfcalc/Calculator.java]: even if u switch around the %s and %d in the header string?

Comment (Riley31415) [reply src/main/java/io/github/seggan/sfcalc/Calculator.java]: idk ill try later

**Review by Seggan — APPROVED** [review]
Looks good 👍

**Pull request merged**